### DISCUSSION TITLE
Trying to group workspaces across monitors

### DISCUSSION THREAD

**Discussion by jrebs** [i3/i3#6241] [Q&A] [answered]
I use a dual monitor setup with i3wm running on an xubuntu machine. My ultimate objective is to try to make i3 behave similarly to have I've traditionally managed virtual workspaces where a workspace covers both displays and when I cycle workspaces or change to a target workspace, both displays are switched accordingly.

I've closely approximated this desire already in i3 by fixing alternating workspaces to specific displays like so:

```
set $ws1 "1:1L"
set $ws2 "2:1R"
set $ws3 "3:2L"
set $ws4 "4:2R"

set $left "DP-4"
set $right "DP-0"

# fix workspaces to particular monitors
workspace $ws1 output $left
workspace $ws2 output $right
workspace $ws3 output $left
workspace $ws4 output $right

# switch to workspace
bindsym $mod+1 workspace $ws1;workspace $ws2
bindsym $mod+2 workspace $ws3;workspace $ws4
```

With this setup, when I use the $mod+1 and $mod+2 keybind, both displays switch to the appropriate workspace (1L+1R, 2L+2R, etc). This is almost good enough for my needs except one problem:

I also use rofi (a launcher/switcher application that I access with a keystroke) and with rofi I like to quickly choose a specific window that I know is running somewhere and jump directly to that window. When I do this, however, i3 obviously falls out of sync and it only changes the workspace on the display that contains the target application to be focused.

Can anyone suggest some practical solution (that doesn't require modifying the i3 source which is beyond me) that would allow me to add some kind of conditional to deal with this?

Ultimately in fake code I want to do something like this:

```
if workspace was just changed:
  find get the workspace that is the 'partner' of the newly focused workspace
  switch to the partner workspace
```

**Comment by okraits** [comment, marked as answer]
You can do that using the IPC interface (https://i3wm.org/docs/ipc.html). I suggest to use the i3ipc python library (https://github.com/altdesktop/i3ipc-python) as it is easy to use and provides some good examples.
You need to subscribe to the workspace event and then act accordingly.
